vmm/src/device_manager.rs

```diff
@@ -805,11 +791,20 @@ impl DeviceManager {
         let serial = if serial_config.mode != ConsoleOutputMode::Off {
             // Serial is tied to IRQ #4
             let serial_irq = 4;
-            let interrupt: Box<dyn devices::Interrupt> =
-                Box::new(UserIoapicIrq::new(ioapic.clone(), serial_irq));
+
+            let base_irq: InterruptIndex = 0;
```

Comment:
So you repeat a block like this over and over. How about a `InterruptManager::create_group_from_configs`, that takes a `Vec<InterruptSourceConfig>` using a default base of zero and counting the number of items in the vector. All this code could then be simplified to 

```
let interrupt_group = InterruptManager::create_from_configs(PIN_IRQ, vec![InterruptSourceConfig::LegacyIrq(LegacyIrqSourceConfig { irq: serial_irq })]);
``

Reply:
Well the problem is that `create_group_from_configs` wouldn't fit the MSI case very well.

Reply:
/cc @sameo

Reply:
If we simplify the way we create legacy interrupt source groups (see my proposal for it), then the duplication would be quite smaller.